src/main/java/edu/umich/its/cpm/MigrationController.java

```diff
@@ -646,44 +648,311 @@ private void JSON_response(HttpServletResponse response, String jsonValue,
+								.createFolder(title);
+						log.info("folder " + title + " created.");
+
+						// push the current folder id into the stack
+						containerStack.push(contentUrl);
+						boxFolderIdStack.push(childFolderInfo.getID());
+						log.debug("after stack peek= " + containerStack.peek()
+								+ " " + " container=" + container);
+						log.debug("*******");
+					} catch (BoxAPIException e) {
+						if (e.getResponseCode() == org.apache.http.HttpStatus.SC_CONFLICT) {
+							// 409 means name conflict - item already existed
+							log.info("There is already a folder with name "
+									+ title);
+						}
+					}
+				}
+
+			} else {
+				// files
+				// Call the uploadFile method to upload file to Box.
+				//
+				log.debug("file stack peek= " + containerStack.peek() + " "
+						+ " container=" + container);
+
+				while (!containerStack.empty()
+						&& !container.equals(containerStack.peek())) {
+					// sync pops
+					containerStack.pop();
+					boxFolderIdStack.pop();
+				}
+
+				if (boxFolderIdStack.empty()) {
+					log.info("Cannot find parent folder for file " + contentUrl);
+					break;
+				}
+
+				String fileName = contentUrl.replace(rootFolderPath, "");
+				uploadFile(boxFolderIdStack.peek(), fileName, contentUrl,
+						sessionId, api);
+			}
+		} // for
+		
+		// refresh tokens
+		BoxUtils.refreshAccessAndRefreshTokens(api);
+	}
+
+	/**
+	 * upload files to Box
+	 */
+	private void uploadFile(String boxFolderId, String fileName,
+			String fileUrl, String sessionId, BoxAPIConnection api) {
+		log.info("begin to upload file " + fileUrl + " to box folder "
+				+ boxFolderId);
+		String contentString = "";
+		try {
+			Service service = new Service();
+			Call nc = (Call) service.createCall();
+
+			nc.setTargetEndpointAddress(env.getProperty("ctools.server.url")
+					+ "sakai-axis/ContentHosting.jws");
+
+			nc.removeAllParameters();
+			nc.setOperationName(OPERATION_GET_CONTENT_DATA);
+			nc.addParameter(OPERTAION_GET_CONTENT_DATA_PARAM_SESSIONID,
+					XMLType.XSD_STRING, ParameterMode.IN);
+			nc.addParameter(OPERTAION_GET_CONTENT_DATA_PARAM_RESOURCEID,
+					XMLType.XSD_STRING, ParameterMode.IN);
+			nc.setReturnType(XMLType.XSD_STRING);
+			contentString = (String) nc.invoke(new Object[] { sessionId,
+					fileUrl });
+
+		} catch (Exception e) {
+			log.error(":zipFiles " + fileUrl + " " + e.getMessage());
 		}
 
-		// get box folders json
-		return BoxUtils.getBoxFolders(boxClientId, boxClientSecret);
+		InputStream content = null;
+
+		Base64.Decoder decoder = Base64.getDecoder();
+		content = new ByteArrayInputStream(decoder.decode(contentString));
+
+		int length = 0;
+		byte data[] = new byte[1024 * 10];
+		BufferedInputStream bContent = null;
+		try {
+
+			bContent = new BufferedInputStream(content);
+			BoxFolder folder = new BoxFolder(api, boxFolderId);
+			final String uploadFileName = fileName;
+			folder.uploadFile(bContent, fileName, 1024, new ProgressListener() {
```

Comment:
Replace with:

	// integer value of stream operation buffer size
	private static final int STREAM_BUFFER_CHAR_SIZE = 1024;